src/panels/RetinaCapturePanel.ts

```diff
@@ -0,0 +1,169 @@
+            startCaptureResponse: "",
+            getAllNodesResponse: [],
+        });
+    }
+}
+
+export class RetinaCaptureProvider implements PanelDataProvider<"retinaCapture"> {
+    constructor(
+        readonly kubectl: k8s.APIAvailable<k8s.KubectlV1>,
+        readonly kubeConfigFilePath: string,
+        readonly kubectlVersion: KubectlVersion,
+        readonly clusterName: string,
+        readonly retinaOutput: string,
+        readonly allNodeOutput: string[],
+        readonly captureFolderName: string
+    ) { }
+
+    getTitle(): string {
+        return `Retina Distributed Capture on ${this.clusterName}`;
+    }
+
+    getTelemetryDefinition(): TelemetryDefinition<"retinaCapture"> {
+        return {
+            getAllNodes: false,
+            runRetinaCapture: true,
+            retinaCaptureResult: true,
+        }
+    }
+
+    getInitialState(): InitialState {
+        return {
+            clusterName: this.clusterName,
+            retinaOutput: [this.retinaOutput],
+            allNodes: this.allNodeOutput,
+            selectedNode: "",
+            captureFolderName: this.captureFolderName,
+        };
+    }
+
+    getMessageHandler(): MessageHandler<ToVsCodeMsgDef> {
+        return {
+            runRetinaCapture: (node: string) => this.handleRunRetinaCapture(node),
+            retinaCaptureResult: () => { },
+            getAllNodes: () => { }
+        };
+    }
+
+    private async handleRunRetinaCapture(node: string) {
+        const localCaptureUri = await window.showSaveDialog({
+            defaultUri: Uri.file(this.captureFolderName),
+            saveLabel: "Download",
+            title: "Download Retina File",
+        });
+
+        if (!localCaptureUri) {
+            return;
+        }
+
+        const localCpPath = getLocalKubectlCpPath(localCaptureUri);
+
+        const createPodYaml = `
+apiVersion: v1
+kind: Pod
+metadata:
+  name: node-explorer-${node}
+spec:
+  nodeName: ${node}
+  tolerations:
+    - key: CriticalAddonsOnly
+      operator: Exists
+    - effect: NoExecute
+      operator: Exists
+    - effect: NoSchedule
+      operator: Exists
+  volumes:
+  - name: mnt-captures
+    hostPath:
+      path: /mnt/capture
+  containers:
+  - name: node-explorer
+    image: alpine
+    command: ["sleep", "9999999999"]
+    volumeMounts:
+    - name: mnt-captures
+      mountPath: /mnt/capture
+`  ;
+
+        const applyResult = await longRunning(`Deploying pod to capture ${node} retina data.`, async () => {
+            return await withOptionalTempFile(createPodYaml, "YAML", async (podSpecFile) => {
+                const command = `apply -f ${podSpecFile}`;
+                return await invokeKubectlCommand(this.kubectl, this.kubeConfigFilePath, command);
+            });
+        });
+
+        if (failed(applyResult)) {
+            vscode.window.showErrorMessage(`Failed to apply Pod: ${applyResult.error}`);
+            throw new Error(`Failed to apply Pod: ${applyResult.error}`);
```

Comment:
Here we're throwing an error in the event of something failing, but a few lines down we're calling `window.showErrorMessage(...)` and returning. I think the latter is more appropriate.

Another option would be to return results (including errors) to the webview. This would also allow the webview to present a link to the capture file, but that gets into a larger discussion which I guess we can postpone for now.